src/runtime_src/core/common/query_requests.h

```diff
@@ -2707,6 +2726,66 @@ struct heartbeat_stall : request
   get(const device*) const = 0;
 };
 
+struct aim_counter : request
+{
+  using result_type = std::vector<uint64_t>;
+  using debug_ip_data_type = debug_ip_data*;
+  static const key_type key = key_type::aim_counter;
+
+  virtual boost::any
+  get(const xrt_core::device* device, const boost::any& arg1) const = 0;
+};
+
+struct am_counter : request
+{
+  using result_type = std::vector<uint64_t>;
+  using debug_ip_data_type = debug_ip_data*;
+  static const key_type key = key_type::am_counter;
+
+  virtual boost::any
+  get(const xrt_core::device* device, const boost::any& arg1) const = 0;
+};
+
+struct asm_counter : request
+{
+  using result_type = std::vector<uint64_t>;
+  using debug_ip_data_type = debug_ip_data*;
+  static const key_type key = key_type::asm_counter;
+
+  virtual boost::any
+  get(const xrt_core::device* device, const boost::any& arg1) const = 0;
+};
+
+struct lapc_status : request
+{
+  using result_type = std::vector<uint32_t>;
+  using debug_ip_data_type = debug_ip_data*;
+  static const key_type key = key_type::lapc_status;
+
+  virtual boost::any
+  get(const xrt_core::device* device, const boost::any& arg1) const = 0;
+};
+
+struct spc_status : request
+{
+  using result_type = std::vector<uint32_t>;
+  using debug_ip_data_type = debug_ip_data*;
+  static const key_type key = key_type::spc_status;
+
+  virtual boost::any
+  get(const xrt_core::device* device, const boost::any& arg1) const = 0;
+};
+
+struct accel_deadlock_status : request
+{
+  using result_type = uint32_t;
+  using debug_ip_data_type = debug_ip_data*;
+  static const key_type key = key_type::accel_deadlock_status;
+
+  virtual boost::any
+  get(const xrt_core::device* device, const boost::any& arg1) const = 0;
+};
```

Comment:
As generic as this is not quite possible unless you create a few domain specific base classes and it would obscure the usage IMO.  E.g. I know I need some device data from sysfs and I want and I know this is a query request.  I would like to understand what query request returns, how it formats the data, what argument it takes etc; all very specific to the request itself and easier to read within a self contained struct.   I agree with the key though.

I definitely see that a comment regarding what 'arg1' is is missing.

Reply:
I have renamed "arg1" as "dbg_ip_data" to indicate that debug_ip_data object is expected to be passed